### PR TITLE
LSIF: update docs

### PR DESCRIPTION
**Pull request by chrismwendt**
This is a continuation of GitHub token auth for LSIF uploads: https://github.com/sourcegraph/sourcegraph/pull/6113

I also removed a note about the old 3.9 release.

### CODE REVIEW

File: doc/user/code_intelligence/lsif.md [hunk 1]
```diff
@@ -61,7 +63,7 @@ When LSIF data does not exist for a particular file in a repository, Sourcegraph
 
 ## Stale code intelligence
 
-LSIF code intelligence will be out-of-sync when you're viewing a file that has changed since the LSIF data was uploaded. You can mitigate this by setting up a periodic job that generates and uploads LSIF for the tip of your default branch (e.g. master) daily. Improvements to this are planned for Sourcegraph 3.9.
+LSIF code intelligence will be out-of-sync when you're viewing a file that has changed since the LSIF data was uploaded. You can mitigate this by setting up a periodic job that generates and uploads LSIF for the tip of your default branch (e.g. master) daily.
```
Comment (ryan-blunden) [inline doc/user/code_intelligence/lsif.md]: Can we go further to say something like "adding a post-build step that uploads LSIF data is the most common workflow"?

Comment (chrismwendt) [reply doc/user/code_intelligence/lsif.md]: Added a "recommended setup" section.

**Comment by chrismwendt** [comment]
> I'd like to see a section about which branches LSIF data should/can be uploaded for, and what pros, cons, and limitations apply.

Added a "recommended setup" section.

File: doc/user/code_intelligence/lsif.md [hunk 2]
```diff
@@ -59,9 +61,15 @@ After uploading LSIF files, your Sourcegraph instance will use these files to po
 
 When LSIF data does not exist for a particular file in a repository, Sourcegraph will fall back to out-of-the-box code intelligence.
 
+## Recommended setup
+
+Start with a periodic job (e.g. daily) in CI that generates and uploads LSIF data on the default branch for your repository.
+
+If you're noticing a lot of stale code intel between LSIF uploads or your CI doesn't support periodic jobs, you can set up a CI job that runs on every commit (including branches). The downsides to this are: more load on CI, more load on your Sourcegraph instance, and more rapid decrease in free disk space on your Sourcegraph instance.
```
Comment (ryan-blunden) [inline doc/user/code_intelligence/lsif.md]: What is the cleanup process for stale LSIF data uploads? Do they have to be managed/deleted manually?

Comment (chrismwendt) [reply doc/user/code_intelligence/lsif.md]: There is no cleanup process right now, filed https://github.com/sourcegraph/sourcegraph/issues/6148 for discussion.

Comment (ryan-blunden) [reply doc/user/code_intelligence/lsif.md]: Adding a `Cleaning up LSIF uploads` section with the above content `There is no cleanup process right now, filed #6148 for discussion.` I think would be useful if we're confident customers will hit this issue.

**Pull request merged**